### PR TITLE
[d3d9] Add VkInterop interfaces

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
Should provide everything needed for VR.

### CODE REVIEW

**Comment by Joshua-Ashton** [comment]
cc: @TheIronWolfModding

File: src/d3d9/d3d9_interfaces.h [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+   * Should be called immediately after submitting
+   * Vulkan commands to the rendering queue in order
+   * to allow DXVK to submit new commands.
+   */
+  virtual void STDMETHODCALLTYPE ReleaseSubmissionQueue() = 0;
+
+  /**
+   * \brief Locks the device
+   * 
+   * Can be called to ensure no D3D9 device methods
```
Comment (K0bin) [inline src/d3d9/d3d9_interfaces.h:161]: You should probably mention that this doesn't do anything if the device wasn't created with the MULTITHREAD flag.

File: src/d3d9/d3d9_interfaces.h [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+   * Executes an explicit image layout transition on the
+   * D3D device. Note that the image subresources \e must
+   * be transitioned back to its original layout before
+   * using it again from D3D9.
+   * \param [in] pTexture The image to transform
+   * \param [in] pSubresources Subresources to transform
+   * \param [in] OldLayout Current image layout
+   * \param [in] NewLayout Desired image layout
+   */
+  virtual void STDMETHODCALLTYPE TransitionTextureLayout(
```
Comment (K0bin) [inline src/d3d9/d3d9_interfaces.h:121]: You should add a note clarifying that the caller is expected to synchronize the device.
Same thing for most of the other functions.

**Review by K0bin — CHANGES_REQUESTED** [review]
Looks mostly fine, just some small nits wrt locking.

**Pull request merged**